src/Components/Main/Main.js

```diff
@@ -0,0 +1,33 @@
+import { Redirect } from 'react-router-dom';
+import { UserContext } from '../../context/UserContext';
+import { useContext, useState } from 'react';
+import './Main.css';
+import { insertToDoItem } from '../../services/client';
+import ToDoItem from '../ToDoItem/ToDoItem';
+import { useToDoItems } from '../../hooks/useToDoList';
+
+export default function Main() {
+  const { user } = useContext(UserContext);
+  const [toDoItem, setToDoItem] = useState('');
+  const [submit, setSubmit] = useState(false);
+  const { toDoList } = useToDoItems(submit);
+  
+  if (!user) {
+    return <Redirect to='/auth/sign-up' />;
+  }
+  return (
+    <div>
+      <label>Add Item to To Do List
+        <input value={toDoItem} onChange={e => setToDoItem(e.target.value)}></input>
+      </label>
+      <button onClick={async () => {
+        await insertToDoItem(toDoItem);
+        setToDoItem('');
+        setSubmit(!submit);
```

Comment:
This is a case where you're setting your state to something that relies on the value of the existing state. This is when you would want to use a callback with the previous state.

```js
setSubmit((prevState) => !prevState)